recipes/cryptopp/all/conanfile.py

```diff
@@ -0,0 +1,75 @@
+from conans import ConanFile, CMake, tools
+import os
+import shutil
+
+
+class CryptoPPConan(ConanFile):
+    name = "cryptopp"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/weidai11/cryptopp"
+    license = "BSL-1.0"
+    description = "Crypto++ Library is a free C++ class library of cryptographic schemes."
+    topics = ("conan", "cryptopp", "crypto", "cryptographic", "security")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, 'fPIC': True}
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        archive_file = 'CRYPTOPP_%s' % self.version.replace('.', '_')
+        os.rename("cryptopp-%s" % archive_file, self._source_subfolder)
+        tools.download("https://raw.githubusercontent.com/noloader/cryptopp-cmake/702bb894f6636e71278d822f01a530397f78966e/CMakeLists.txt",
+                       os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                       sha256="e2268878bc3b779334dd57515d7656f0567b879a6c8dcfb876cabbe72dccbac1")
```

Comment:
It would be better to move this in conandata:

```yml
sources:
  "8.2.0":
    url:
      "source": "https://github.com/weidai11/cryptopp/archive/CRYPTOPP_8_2_0.tar.gz"
      "cmake": "https://github.com/noloader/cryptopp-cmake/archive/CRYPTOPP_8_2_0.tar.gz"
    sha256:
      "source": "e3bcd48a62739ad179ad8064b523346abb53767bcbefc01fe37303412292343e"
      "cmake": "e2268878bc3b779334dd57515d7656f0567b879a6c8dcfb876cabbe72dccbac1"
```